test/common/memory/heap_shrinker_test.cc

```diff
@@ -24,7 +24,7 @@ class HeapShrinkerTest : public testing::Test {
       : api_(Api::createApiForTest(stats_, time_system_)), dispatcher_(*api_, time_system_) {}
 
   void step() {
-    time_system_.sleep(std::chrono::milliseconds(10000));
+    time_system_.advanceTime(std::chrono::milliseconds(10000));
     dispatcher_.run(Event::Dispatcher::RunType::NonBlock);
```

Comment:
Same here - I assume the non-block is to give time for the callback to run.  I guess I'm not sure where we'd advance and not run alarms, and I wonder how many of these cases are holdover from the differences not clearly being spelled out?  Most of these tests are doing a "walk forward" and make sure my thing is called" check.
Is it just that pending alarms is not the right list?  I had assumed pending alarms were the alarms which are due to fire and have not yet fired.  I think I need to understand this PR better before reviewing.

Reply:
there's no other thread in this test; the callbacks are run by the dispatcher_.run() call below. If we called sleep() here it would deadlock since no thread would be alive to run the callbacks.